На ветке lesson10

RamzanBersanov opened the pull request:
Изменения, которые будут включены в коммит:
	новый файл:    .vscode/settings.json
	новый файл:    app/assets/stylesheets/flash.scss
	новый файл:    app/assets/stylesheets/nav.scss
	изменено:      app/controllers/application_controller.rb
	изменено:      app/controllers/questions_controller.rb
	новый файл:    app/controllers/sessions_controller.rb
	изменено:      app/controllers/test_passages_controller.rb
	новый файл:    app/controllers/users_controller.rb
	новый файл:    app/helpers/sessions_helper.rb
	новый файл:    app/helpers/users_helper.rb
	изменено:      app/models/user.rb
	новый файл:    app/models/user/auth.rb
	изменено:      app/views/layouts/application.html.erb
	новый файл:    app/views/sessions/_form.html.erb
	новый файл:    app/views/sessions/new.html.erb
	новый файл:    app/views/shared/_nav.html.erb
	новый файл:    app/views/users/_form.html.erb
	новый файл:    app/views/users/new.html.erb
	изменено:      config/routes.rb
	новый файл:    db/migrate/20230101154901_add_password_digest_to_users.rb

RamzanBersanov committed:
На ветке lesson10
 Изменения, которые будут включены в коммит:
	новый файл:    .vscode/settings.json
	новый файл:    app/assets/stylesheets/flash.scss
	новый файл:    app/assets/stylesheets/nav.scss
	изменено:      app/controllers/application_controller.rb
	изменено:      app/controllers/questions_controller.rb
	новый файл:    app/controllers/sessions_controller.rb
	изменено:      app/controllers/test_passages_controller.rb
	новый файл:    app/controllers/users_controller.rb
	новый файл:    app/helpers/sessions_helper.rb
	новый файл:    app/helpers/users_helper.rb
	изменено:      app/models/user.rb
	новый файл:    app/models/user/auth.rb
	изменено:      app/views/layouts/application.html.erb
	новый файл:    app/views/sessions/_form.html.erb
	новый файл:    app/views/sessions/new.html.erb
	новый файл:    app/views/shared/_nav.html.erb
	новый файл:    app/views/users/_form.html.erb
	новый файл:    app/views/users/new.html.erb
	изменено:      config/routes.rb
	новый файл:    db/migrate/20230101154901_add_password_digest_to_users.rb

VidgarVii commented on .vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
файлы настройки редактора кода к проекту не относятся. Не стоит их класть в репозиторий

RamzanBersanov replied:
я не знаю, как не класть в репозиторий, то что есть в проекте локально

VidgarVii replied:
коммитеть можно только определенные файлы, а не всё сразу

Чтобы папка не попадала в индекс надо её прописать в gitignore

VidgarVii commented on app/assets/stylesheets/flash.scss:
```diff
@@ -0,0 +1,9 @@
+  
+  .flash.alert {
+    border-color: #ef2727;
+  }
```
No new lina at end of file

VidgarVii commented on app/assets/stylesheets/nav.scss:
```diff
@@ -0,0 +1,3 @@
+.nav.user {
+    text-align: right;
+  }
```
No new lina at end of file

VidgarVii commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,24 @@
+      current_user.present?
+    end
+  
+  end
```
No new lina at end of file

VidgarVii commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,27 @@
+        if cookies[:users_initial_request] != nil 
+          redirect_to cookies[:users_initial_request] 
+        else 
+          redirect_to tests_path 
```
очень сложный код получился. Если в коде есть вложенные if else значит что-то делаешь не так.
Если кук нет то можно использовать оператор ||
`redirect_to cookies[:users_initial_request] || tests_path`
метод delete вернёт то, что удаляешь. По этому удалять можно сразу при редиректе
`redirect_to cookies.delete(:users_initial_request) || tests_path`

VidgarVii commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,27 @@
+    end
+  
+  end
+  
```
No new lina at end of file

RamzanBersanov committed:
На ветке lesson10
 Изменения, которые будут включены в коммит:
	изменено:      app/assets/stylesheets/flash.scss
	изменено:      app/assets/stylesheets/nav.scss
	изменено:      app/controllers/application_controller.rb
	изменено:      app/controllers/sessions_controller.rb
	изменено:      app/controllers/tests_controller.rb

RamzanBersanov committed:
На ветке lesson10
 Изменения, которые будут включены в коммит:
	изменено:      app/controllers/tests_controller.rb
	изменено:      app/controllers/users_controller.rb

VidgarVii commented on app/assets/stylesheets/flash.scss:
```diff
@@ -0,0 +1,10 @@
+  .flash.alert {
+    border-color: #ef2727;
+  }
+  
```
No new lina at end of file не были убраны

VidgarVii replied:
не исправлено по всему проекту

VidgarVii commented on app/controllers/tests_controller.rb:
```diff
@@ -42,9 +43,8 @@ def destroy
   end
```
тест не создаётся

RamzanBersanov committed:
На ветке lesson10
 Изменения, которые будут включены в коммит: создание теста, гитигнор, пустые строки в конце файлов

RamzanBersanov committed:
Delete .vscode directory

VidgarVii commented on app/controllers/tests_controller.rb:
```diff
@@ -42,15 +40,14 @@ def destroy
 
   def test_params
-    params.require(:test).permit(:title, :level, :category_id)
+    params.require(:test).permit(:title, :level, :category_id, :author_id)
```
автор это текущий пользователь. Не стоит передавть author_id через форму

VidgarVii commented on app/views/shared/_nav.html.erb:
```diff
@@ -0,0 +1,13 @@
+
+<% if logged_in? %>
+    <p class="nav user">Добро пожаловать, <%= current_user.name %></b>Гуру!</p>
+    <%= link_to "Выйти из аккаунта", session_path(session), method: :delete %>
```
с 6 рельсы такой подход больше не работает

Commit:
On branch lesson10
автор айди и пустые строки#
